packages/typescript-to-proptypes/src/generator.ts

```diff
@@ -269,6 +269,10 @@ export function generate(component: t.Component, options: GenerateOptions = {}):
         .reduce((prev, curr) => `${prev},${curr}`)}])${isOptional ? '' : '.isRequired'}`;
     }
 
+    if (propType.type === 'UndefinedNode' && context.propTypeDefinition.name === 'variant') {
+      return `${importedName}.any`;
+    }
+
```

Comment:
Have to update the generator to make an exception for `variant` prop. We never encounter this scenario before where the prop is intended to be augmented for styling.

Need another review, cc @mnajdova

Reply:
@siriwatknp Sorry, I accidentally removed your review requests. Could you please request again from the correct people? This has been a blocker for 2 months now at our company.

Reply:
It will be useful to https://github.com/mui/material-ui/issues/34238 as well.